core/modules/filter/filter.admin.inc

```diff
@@ -376,10 +409,178 @@ function filter_admin_format_form($form, &$form_state, $format) {
+    filter_clear_format_tempstore($format->format);
+  }
+  $form_state['redirect'] = 'admin/config/content/formats';
+}
+
+/**
+ * Validation handler for the settings configuration buttons on the
+ * filter_admin_format_form().
+ */
+function filter_admin_format_filter_settings_form_redirect_validate($form, &$form_state) {
+  // Throw a form error if there is no machine name yet, as we need this to set
+  // up the URLs for settings forms.
+  if (empty($form_state['values']['format'])) {
+    form_set_error('format', t('Please enter a format name before configuring settings'));
+  }
+}
+
+/**
+ * Submit handler for the settings configuration buttons on the
+ * filter_admin_format_form().
+ */
+function filter_admin_format_filter_settings_form_redirect($form, &$form_state) {
+  $trigger = $form_state['triggering_element'];
+  $name = $trigger['#name'];
+  if (empty($form_state['format']->format)) {
+    $form_state['format']->format = $form_state['values']['format'];
+    filter_set_format_tempstore($form_state['format']);
+  }
+  $form_state['redirect'] = 'admin/config/content/formats/' . $form_state['format']->format . '/filter-settings/' . $name;
+}
+
+/**
+ * Ajax handler: Handles multistep buttons in field_ui_display_overview_form().
+ */
+function filter_admin_format_configure_ajax($form, &$form_state) {
+  $trigger = $form_state['triggering_element'];
+  $name = $trigger['#name'];
+  $format = $form_state['format'];
+  $format = filter_get_format_tempstore($format->format);
+
+  // Transfer alllowed tags from the hidden textfield to the format object in 
+  // case there were button changes which affected allowed tags.
+  $format->filters['filter_html']->settings['allowed_html'] = $form_state['values']['allowed_html'];
+
+  $configure_form = backdrop_get_form('filter_admin_format_filter_settings_form', $format, $name);
+  $title = t('Configure filter');
+  $options = array(
+    'dialogClass' => 'filter-dialog',
+    'width' => '90%',
+  );
+
+  $html = '';
+  $html .= theme('status_messages');
+  $html .= '<div data-filter-name="' . $name . '" class="filter-dialog-settings">';
+  $html .= backdrop_render($configure_form);
+  $html .= '</div>';
+
+  $commands = array();
+  $commands[] = ajax_command_open_modal_dialog($title, $html, $options);
+
+  return array('#type' => 'ajax', '#commands' => $commands);
+}
+
+/**
+ * Ajax handler for buttons in filter_admin_format_filter_settings_form().
+ */
+function filter_admin_format_configure_ajax_submit($form, &$form_state) {
+  $commands = array();
+  if (isset($form_state['values']['allowed_html'])) {
+    $commands[] = ajax_command_data('#allowed-html', 'copyAllowedVals', $form_state['values']['allowed_html']);
+    $selector = '#edit-filters-' . str_replace('_', '-', $form_state['filter']) . '-filter .filter-details.description';
+    $replacement = check_plain('Allowed HTML tags: ' . $form_state['values']['allowed_html']);
+    $commands[] = ajax_command_html($selector, $replacement);
+  }
```

Comment:
In addition to updating the description, you could update the hidden `#allowed-tags` element value as an AJAX command here.
```
$commands[] = ajax_command_invoke('#allowed-html', 'val', $form_state['values']['allowed_html']);
$commands[] = ajax_command_invoke('#allowed-html', 'trigger', 'change');
```